plugins/GSdx/GSThread_CXX11.h

```diff
@@ -24,6 +24,87 @@
 #include "GSdx.h"
 #include "boost_spsc_queue.hpp"
 
+#define QUEUE_SEM
+
+#ifdef QUEUE_SEM
+
+#include <semaphore.h>
+#include <errno.h> // EBUSY
+#include <pthread.h>
+
+template<class T, int CAPACITY> class GSJobQueue final
+{
+private:
+	std::thread m_thread;
+	std::function<void(T&)> m_func;
+	std::atomic<bool> m_exit;
+	std::atomic<int32_t> m_count;
+	ringbuffer_base<T, CAPACITY> m_queue;
+
+    sem_t m_sem_work;
+
+	void ThreadProc() {
+		while (true) {
+			sem_wait(&m_sem_work);
+
+			if (m_exit.load(memory_order_relaxed)) {
+				return;
+			}
+
+			m_queue.consume_one(*this);
+
+			m_count--;
+		}
+	}
+
+public:
+	GSJobQueue(std::function<void(T&)> func) :
+		m_func(func),
+		m_exit(false)
+	{
+		m_thread = std::thread(&GSJobQueue::ThreadProc, this);
+		sem_init(&m_sem_work, false, 0);
```

Comment:
Well, you're calling sem_wait immediately in ThreadProc. It'll fix the crashing.